### DISCUSSION TITLE
Zero Export setting

### DISCUSSION THREAD

**Discussion by Zigbee66** [ksheumaker/homeassistant-apsystems_ecur#195] [Q&A]
Hi Everyone,

Does anybody know if there’s a way to control the Zero Export functionality of the ECU-C via Home Assistant (or anything else in fact).

Thanks in advance, all the best:

Eric

**Comment by hautamik** [comment]
Hi! 

I just last week had my solar panel system installed, and went looking for this same thing. Unfortunately I couldn't find an answer, so I thought I'd try something by myself.

Loaded ECU-C's web interface page where this Zero Export can be activated/de-activated. Said "Inspect" in Firefox and found the form and script related to the form's variables. URL, method and parameters can be read there.

```javascript
if(check_error == 0)
    {
    	$.ajax({
			url : "http://192.168.1.165/index.php/meter/set_meter_display_funcs",
			type : "post",
        	dataType : "json",
			data: "&meter_func=" + $("#meter_func").val()
				+ "&this_func=1" 
				+ "&power_limit=" + $("#power_limit").val()*1000,
	    		success : function(Results){
	    			$("#result").text(Results.message);
            	if(Results.value == 0){
	                	$("#result").removeClass().addClass("alert alert-success");
                	setTimeout('$("#result").fadeToggle("slow")', 3000);
	            	}
            	else{
                	$("#result").removeClass().addClass("alert alert-warning");
            	}              		 
            	$("#result").fadeToggle("slow");
        		window.scrollTo(0,0);//页面置顶
            	setTimeout("location.reload();",60000);//刷新页面
			},
	    		error : function() { alert("Error");}
    	})   
    }    
``` 

Fill form, click Save and Firefox also shows the needed header information. With a little help from Postman, I was able to create curl command to test this. Seems to be working. 

Set Zero Export (meter_func=1)

If you intend to set something else than zero watts, put power_limit in watts, as the form asks the limit in kW and multiplies the parameter by 1000, so 1 kW -> 1000 (watts).

```shell
curl --location 'http://[ECU-C IP ADDRESS]/index.php/meter/set_meter_display_funcs' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'meter_func=1' \
--data-urlencode 'this_func=1' \
--data-urlencode 'power_limit=0'
``` 

Unset Zero Export (meter_func=0)

```shell
curl --location 'http://[ECU-C IP ADDRESS]/index.php/meter/set_meter_display_funcs' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'meter_func=0' \
--data-urlencode 'this_func=1' \
--data-urlencode 'power_limit=0'
```
